src/apps/vhost/vhost_user.lua

```diff
@@ -86,6 +100,28 @@ function VhostUser:push ()
    end
 end
 
+function VhostUser:tx_callback (p)
+   counter.add(self.counters['out-octets'], packet.length(p))
+   if ethernet:is_mcast(packet.data(p)) then
+      counter.add(self.counters['out-multicast'])
+   else
+      counter.add(self.counters['out-unicast'])
+   end
+end
+
+function VhostUser:rx_callback (p)
+   counter.add(self.counters['in-octets'], packet.length(p))
+   if ethernet:is_mcast(packet.data(p)) then
+      counter.add(self.counters['in-multicast'])
+   else
+      counter.add(self.counters['in-unicast'])
+   end
+end
```

Comment:
Is it safe to introduce this potentially unbiased branch onto the "fast path" for Virtio-net?

The risk I see is that on a workload with 50/50 mix of unicast/multicast traffic you will take the penalty of both a LuaJIT side-trace and also a CPU branch-misprediction half of the time. This could be significant and we don't currently have performance test coverage for such a workload.

One alternative would be to write this branch-free (using arithmetic, bitwise operators, and min/max). Sketch:

```lua
-- Set unicast and multicast to 0 and 1 as appropriate.
local multicast = bit.band(packet.data[0], 1)
local unicast = 1 - multicast
counter.add(self.counters['in-multicast'], multicast)
counter.add(self.counters['in-unicast'], unicast)
```

This way the same instructions would execute every time and only the values would change.